sdk/trace/sampling.go

```diff
@@ -39,16 +39,18 @@ type SamplingDecision struct {
 	Sample bool
 }
 
-// ProbabilitySampler returns a Sampler that samples a given fraction of traces.
-//
-// It also samples spans whose parents are sampled.
+// ProbabilitySampler samples a given fraction of traces. Fractions >= 1 will
+// always sample. If the parent span is sampled, then it's child spans will
+// automatically be sampled. Fractions <0 are treated as zero, but spans may
+// still be sampled if their parent is.
 func ProbabilitySampler(fraction float64) Sampler {
-	if !(fraction >= 0) {
-		fraction = 0
-	} else if fraction >= 1 {
+	if fraction >= 1 {
 		return AlwaysSample()
 	}
 
+	if fraction <= 0 {
+		fraction = 0
```

Comment:
I think we could change this to use `NeverSample` in the same way when `fraction >= 1` uses `AlwaysSample`?

Reply:
I would leave it as fraction = 0. The reason is if you want to sample if parent is sample but not otherwise then probability sampler with 0 value will do the job.

Reply:
You're right  @rghetia. Seeing the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-tracing.md#built-in-samplers), we are missing the `ALWAYS_PARENT` sampler, which could be used in this case. But this is out of the scope of this issue.